ultravox/data/datasets.py

```diff
@@ -266,7 +267,32 @@ def _get_messages(
     return messages
 
 
-class VoiceDataset(abc.ABC, data.IterableDataset):
+class SizedIterableDataset(data.IterableDataset):
+    """
+    A wrapper for a Dataset that provides a length method and verifies the count.
+    """
+
+    def __init__(self, dataset: data.Dataset, estimated_length: int) -> None:
+        self._dataset = dataset
+        self._estimated_length = estimated_length
+
+    def __iter__(self):
```

Comment:
When is this function ever used?
From what I see this is supposed to be a superclass of `VoiceDataset` which already has an `__iter__` function so it gets overwritten.

Reply:
I see this class is used both as a parent class and as a wrapper. That solves this issue I guess ...

Reply:
Changed the class into an interface, and implemented the len and iter in the voicesample class.